File: 백준/Silver/18870. 좌표 압축/좌표 압축.cs
using System;
using System.Collections.Generic;
using System.Linq;

class Program
{
    static void Main()
    {
        int number = int.Parse(Console.ReadLine());
        int[] array = Console.ReadLine().Split(' ').Select(int.Parse).ToArray();
        
        int[] uniqueArray = array.Distinct().ToArray();
        Array.Sort(uniqueArray);
        
        Dictionary<int, int> myMap = new Dictionary<int,int>();
        for (int i=0; i<uniqueArray.Length; i++)
        {
            myMap[uniqueArray[i]] = i;
        }
        
        string answer = string.Join(" ", array.Select(x=>myMap[x]));
        
        Console.WriteLine(answer);
    }
}

File: 백준/Silver/18870. 좌표 압축/좌표 압축.js
let input = require('fs')
.readFileSync('/dev/stdin')
.toString().split('\n');

let n = Number(input[0]);
let array = input[1].split(' ').map(Number);

let uniqueArray = [...new Set(array)];
uniqueArray.sort(function(a,b){
        return a - b
});

let myMap = new Map();
for (let i=0; i<uniqueArray.length; i++)
{
    myMap.set(uniqueArray[i],i);
}
answer = "";
for(x of array){
    answer += myMap.get(x) + " ";
}

console.log(answer);